src/amm/amm.validators.ts

```diff
@@ -309,6 +338,66 @@ export const validateRemoveLiquidityRequest: RequestValidator =
     validateMaxPriorityFeePerGas,
   ]);
 
+export const validateCosmosPriceRequest: RequestValidator = mkRequestValidator([
```

Comment:
Removing the cosmos specific controllers will allow you should remove these validators as well. This folder should only contain generic AMM code.

Reply:
How would you want to handle the non-required vars then, just validate if not null? Then Uniswap will need to perform not-null checks inside its controllers.

Reply:
There shouldn't be much changes needed to the validators. If you make the params optional as in:
```
export interface PoolPriceRequest extends NetworkSelectionRequest {
  token0: string;
  token1: string;
  fee?: string;
  period?: number;
  interval?: number;
}
```

The `mkRequestValidator` function in the validators in `amm.validators.ts` will only test the required params:
```
export const validatePoolPriceRequest: RequestValidator = mkRequestValidator([
  validateConnector,
  validateChain,
  validateNetwork,
  validateToken0,
  validateToken1,
  validateFee,
  validateInterval,
  validatePeriod,
]);
```

This may cause validation issues for connectors that require certain arguments in this request, such as `fee` for Uniswap. If that's needed, the Uniswap connector should validate the presence of that argument in its dedicated controllers in its folder.